Decouple island location from island center.

tastybento opened the pull request:
This PR enables the protection area to move anywhere within the island boundaries. This is required to enable a new game mode I'm working on where the island protection zone starts small and increases based on achievements/challenges. I need a way to move the protection zone anywhere within the island boundaries and not be locked to the center.

I added an admin command to enable manual setting of the island location. I also expanded the info command to show the island location.

In summary, the island location is the center of the protection area. It is usually the center of the island but can be moved. The protection area will not go outside the island range. However, as it is possible for the island location to be right on the edge of the island range, the protection range value can now be up to the diameter of the island, i.e., 2 * island range.

tastybento committed:
Decouple island location from island center.

This enables the protection area to move anywhere within the island
boundaries

tastybento committed:
Paste island at island location.

tastybento committed:
Move visitors to the island location

tastybento committed:
Added island location placeholders

tastybento committed:
Use island location instead of island center

BONNe commented on src/main/java/world/bentobox/bentobox/listeners/JoinLeaveListener.java:
```diff
@@ -205,8 +205,8 @@ private void updateIslandRange(User user) {
-                // Range cannot be greater than the island distance
-                range = Math.min(range, plugin.getIWM().getIslandDistance(island.getWorld()));
+                // Range cannot be greater than the island distance * 2
+                range = Math.min(range, 2 * plugin.getIWM().getIslandDistance(island.getWorld()));
```
I am not sure if this is a correct way how to do it.

I would prefer to calculate by offset from the actual island center.

tastybento replied:
The protection range is now relative to the protection center location, so if the player is to be given the ability to use all of their island area, then their protection range must be able to go up to 2x island distance away from that point.

For example:

Island range = 400
Island center = 0,0
Island location = 399, 0
Island protection range = 800

This last value will enable them to operate anywhere with all of the island space.

tastybento committed:
Fix tests

tastybento committed:
Allow radius 1 islands.

tastybento committed:
Correct admin range set limit

tastybento committed:
Debug in progress

tastybento committed:
Fix island deletion.

The speed option was doing nothing except causing repeated actions on
the same chunks.

tastybento committed:
Soft depend for WorldGeneratorApi for Boxed game mode

tastybento committed:
Merge remote-tracking branch 'origin/develop' into island_location

Conflicts:
	src/main/java/world/bentobox/bentobox/util/DeleteIslandChunks.java

Poslovitch reviewed (approved):
Everything looks good. Do not forget to rename the placeholders and the commands appropriately, to keep up with the suggested change.

Thanks for your work, and sorry for the time it took me to finally leave a review.

tastybento committed:
Changed to "protectionCenter".

Added logic to expand the deletion area based on moving the island
protection location and range instead of just deleting everything. This
will keep deletion time to a minimum for BSkyBlock, etc.

tastybento committed:
Merge branch 'develop' into island_location

Conflicts resolved:
	src/test/java/world/bentobox/bentobox/listeners/PortalTeleportationListenerTest.java

tastybento committed:
Fix tests.
